### ISSUE TITLE
No error message presented when incorrect billing province is used in Apple Pay.

### ISSUE THREAD

**Issue by avantegarde** [woocommerce/woocommerce-gateway-stripe#2569]
**Describe the bug**
Apple Pay returns `Payment Not Completed` when the billing province is incorrect. But there is no error message presented to the customer on the front-end to indicate that the error is with the billing province.

**To Reproduce**
1. Attempt to pay for an order with Apple Pay
2. Use the following address:

```
Viale Eroi Di Cefalonia 145
00128
Roma
Lazio
Italy
```

4. Apple Pay will return `Payment Not Completed` and close the modal.
5. No error is present on the checkout page indicating that the issue was the billing province.

If you instead use the following address, the order will be successful.

```
Viale Eroi Di Cefalonia 145
00128
Roma
Roma
Italy
```

**Expected behavior**
When entering an incorrect billing province in Apple Pay, Stripe should output an error that states that the billing province is invalid.

**Screenshots**
![Screen Shot 2023-02-12 at 4 24 40 PM](https://user-images.githubusercontent.com/3516737/224092940-076cd8a5-b601-4a83-b968-395f21eb5a1a.png)

**Environment (please complete the following information):**
 - WordPress Version: 6.1.1
 - WooCommerce Version: 7.3.0
 - Stripe Plugin Version: 7.1.0

**Additional context**
Additional details and conversation: p1676217694284599-slack-C7U3Y3VMY

WooCommerce Payments lets the customer know that the billing province is not valid:

![Screen Shot on 2023-02-22 at 13-50-37](https://user-images.githubusercontent.com/3516737/224046696-5d5049d3-2d01-41a0-852f-abd0329249e2.png)

Square also lets the customer know that the issue is with the Province:

![image](https://user-images.githubusercontent.com/3516737/224046862-2d85816a-6fdd-4fa9-a0cb-e44183823035.png)

5917996-zd-woothemes

**Comment by pierorocca** [comment]
@avantegarde do you happen to have the UI that's allowing Lazio, a region not a province, to be entered as a province? Or is the issue happening on the back-end?

**Comment by avantegarde** [comment]
My apologies @pierorocca for the delay here. I believe the address is pre-populated by Apple Pay within their shipping address UI.

<img width="409" alt="Immagineallegata-1" src="https://user-images.githubusercontent.com/3516737/232607815-13901841-74e3-4778-b80c-47d835cdfab6.png">

**Comment by paulostp** [comment]
Just adding to @avantegarde's comment. When you start adding an address to Apple Pay, the form will offer to auto-complete it. If you accept it, Lazio will be used instead of Roma.

![1](https://user-images.githubusercontent.com/69197772/232714625-07a3322a-f460-4c34-b98b-304d9bd2b2b5.png)
![2](https://user-images.githubusercontent.com/69197772/232714644-40cb1570-65a0-4bb7-bc1a-4d97bf93bc6c.png)

**Comment by pierorocca** [comment]
Definitely an Apple Pay bug. Lazio is not a province. Google gets it right.
![image](https://user-images.githubusercontent.com/109928953/232861378-97bd3b1b-a38c-4697-88d7-d313e7fc0325.png)

**Comment by pierorocca** [comment]
Let me raise this with the partnerships team that works with Apple.

**Comment by pierorocca** [comment]
Email has been sent to our contacts at Apple highlighting this issue. I understand this has been brought up in the past highlighting Hong Kong addresses as being problematic.

**Comment by mbellagamba** [comment]
I just want to point out that I'm experiencing the same issue. It probably happens with all Italy addresses. As an additional information, I noticed that the payment is successful also removing the incorrect Province (the value "Lazio" in the  @avantegarde's example). However this is a poor experience for users that are already paying using an address configured like that in other websites. Maybe it could be possible to fix it just avoiding to check that value for specific countries.

**Comment by pierorocca** [comment]
@bborman22 with ECE do we have a way of working around this?

**Comment by bborman22** [comment]
Sorry I had missed this! While I think ECE can handle this case, I don't think it's anything particular to ECE. Though to confirm I tested this with ECE on WooPayments with the Apple Pay autocomplete address using Lazio and confirmed I got the following error message displayed:

![Image](https://github.com/user-attachments/assets/58744828-780c-4ac5-800f-7101b07c6e57)

**Comment by pierorocca** [comment]
Yes it's another situation where Apple localization is wrong similar to HK.
